src/acvp_transport.c

```diff
@@ -595,9 +634,19 @@ ACVP_RESULT acvp_retrieve_vector_set_result (ACVP_CTX *ctx, int vs_id) {
     }
     rv = acvp_curl_http_get(ctx, url, &acvp_curl_write_kat_func);
     if (rv != HTTP_OK) {
-        ACVP_LOG_ERR("Unable to get vector result from server. curl rv=%d\n", rv);
-        ACVP_LOG_ERR("%s\n", ctx->kat_buf);
-        return ACVP_TRANSPORT_FAIL;
+        if (rv == HTTP_UNAUTH) {
```

Comment:
Since this code seems mostly common and used in 4 places, what do you think about making a new function "acvp_http_get" that encapsulates these return code checks and the refresh?

Reply:
I would except for some are GETs some are POSTs and there are 3 different write functions would be lots of parameters passed in.